File: backend/core/views/publisher.py
from django_filters import rest_framework as filters
from rest_framework import viewsets
from ..models import Publisher
from ..serializers import PublisherSerializer
from rest_framework.response import Response
from rest_framework import status

class PublisherViewSet(viewsets.ModelViewSet):
    queryset = Publisher.objects.all()
    serializer_class = PublisherSerializer
    filter_backends = [filters.DjangoFilterBackend]
    filterset_fields = ['name','city','country']

    def create(self, request, *args, **kwargs):
        serializer = self.get_serializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        self.perform_create(serializer)
        headers = self.get_success_headers(serializer.data)
        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)

File: frontend/src/app/utils/publisher.ts
export class Publisher {
    id: number;
    name: string;
    address: string;
    city: string;
    country: string;
    website: string;
  
    constructor(id: number, name: string, address: string, city: string, country: string, website: string) {
      this.id = id;
      this.name = name;
      this.address = address;
      this.city = city;
      this.country = country;
      this.website = website;
    }
  
    getLocation() {
      return `${this.city}, ${this.country}`;
    }
  }